src/sync/lookupApiUtils.js

```diff
@@ -193,31 +200,75 @@ const processPrescriberResponse = response => {
 export const queryPatientApi = async params => {
   const requestUrl = getPatientRequestUrl(params);
   const headers = getRequestHeaders();
+  const options = { headers };
   try {
-    const response = await fetch(requestUrl, { headers });
-    const responseJson = await response.json();
-    const { error } = responseJson;
-    if (error) throw new Error(error);
-    const patientData = processPatientResponse(responseJson);
-    return patientData;
+    const response = await fetchTimeout(requestUrl, options, CONNECTION_TIMEOUT_PERIOD);
+    const { ok, status } = response;
+    if (ok) {
+      const responseJson = await response.json();
+      const { error } = responseJson;
+      if (error) throw new Error(error);
+      const patientData = processPatientResponse(responseJson);
+      return patientData?.length
+        ? { data: patientData }
+        : { error: ERROR_CODES.NO_MATCHING_RECORDS };
+    }
+    switch (status) {
+      case 400:
+        throw new Error(ERROR_CODES.CONNECTION_FAILURE);
+      case 401:
+        throw new Error(ERROR_CODES.INVALID_PASSWORD);
+      default:
+        throw new Error(ERROR_CODES.CONNECTION_FAILURE);
+    }
   } catch (error) {
     // TODO: add bugsnag.
-    return [];
+    return { error: error.message };
   }
 };
 
 export const queryPrescriberApi = async params => {
   const requestUrl = getPrescriberRequestUrl(params);
   const headers = getRequestHeaders();
+  const options = { headers };
   try {
-    const response = await fetch(requestUrl, { headers });
-    const responseJson = await response.json();
-    const { error } = responseJson;
-    if (error) throw new Error(error);
-    const prescriberData = await processPrescriberResponse(responseJson);
-    return prescriberData;
+    const response = await fetchTimeout(requestUrl, options, CONNECTION_TIMEOUT_PERIOD);
+    const { ok, status } = response;
+    if (ok) {
+      const responseJson = await response.json();
+      const { error } = responseJson;
+      if (error) throw new Error(error);
+      const prescriberData = await processPrescriberResponse(responseJson);
+      return prescriberData?.length
+        ? { data: prescriberData }
+        : { error: ERROR_CODES.NO_MATCHING_RECORDS };
+    }
+    switch (status) {
+      case 400:
+        throw new Error(ERROR_CODES.CONNECTION_FAILURE);
+      case 401:
+        throw new Error(ERROR_CODES.INVALID_PASSWORD);
+      default:
+        throw new Error(ERROR_CODES.CONNECTION_FAILURE);
+    }
```

Comment:
this code looks eerily familiar! 🤣

Reply:
I think its new? Is it already in the codebase?

Reply:
Oh, you mean the method duplication :laughing:. Yeah, I don't like it either but trying to follow the rule of 2 (only abstract methods if used in more than two places).